backends/v3/src/radix.rs

```diff
@@ -1,10 +1,17 @@
 use crate::block_allocator::{Allocator, BlockAllocation};
 use slotmap::{DefaultKey, SlotMap};
+use std::hash::{Hash, Hasher};
 use std::{
     collections::{BTreeSet, HashMap},
     sync::Arc,
 };
 
+fn hash_(slice: &[u32]) -> u32 {
```

Comment:
The crux of the change. Instead of matching on a single token (input_ids) we match on the entire block content.